### DISCUSSION TITLE
箱庭ドローンシミュレータのARデバイス対応に関して

### DISCUSSION THREAD

**Discussion by yamati-kz** [toppers/hakoniwa#187] [Q&A]
@tmori さん　いつもいろいろとありがとうございます。少々質問させてください。

[ARデバイスの対応](https://github.com/toppers/hakoniwa-px4sim/blob/main/README-ja.md#ar%E3%83%87%E3%83%90%E3%82%A4%E3%82%B9%E3%81%AE%E5%AF%BE%E5%BF%9C)を拝見させていただいております。

1. Oculus Quest 3 または他の対応するARデバイス
   他の対応するARデバイスと[必要な機材](https://github.com/toppers/hakoniwa-px4sim/blob/main/README-ja.md#%E5%BF%85%E8%A6%81%E3%81%AA%E6%A9%9F%E6%9D%90)の中に記載がありますが、どのような条件でしょうか？
   1.1 Android OSおよびAndroidアプリケーションがARデバイス上で走ること。
   1.2 パススルー機能があること。(デバイスによっては、パススルー機能そのものがないものもあるらしい)
   1.3 AR対応デバイスは、ルータに有線接続できること。
         USB-TypeC->LANアダプタをAR対応デバイスに接続する。RJ45にLANケーブルを接続して更にルータ側に接続する。
         https://www.docswell.com/s/kanetugu2015/K4VVJD-hakoniwa-drone-ar-op#p3を参照しています。
2. ルータ性能の影響
    最低限のスペックをご存知であれば、提示していただけると助かります。

お忙しいと思いますので、お手漉きのときにゆっくいで構いません。
よろしくお願いします。

**Comment by tmori** [comment]
> 1. 必要な機材

現時点では、Quest3のみの対応になります。
なので、1.1 は必須ですし、1.2も必須ですね。1.3は、Wifiでもいけますね。

**Reply by yamati-kz** [reply]
> 現時点では、Quest3のみの対応になります。

   はい。この手のものをせっかく入手するなら別機種にして動作検証して対応機種のバリエーションを増やせたらいいかなと思ったりも。
　諸々悩み中です。
   近々発売になる[Quest 3S](https://www.meta.com/jp/quest/quest-3s/)ですしょうか。
   [Pico4 ultra](https://www.picoxr.com/jp/products/pico4-ultra)でもAndroidアプリが動きそうです。搭載OSは調べきれてませんが。

> 1.3は、Wifiでもいけますね。

   デモ映像を撮られたときは、wifiでしょうか？

話は変わって。
スマホも必要ですね。Quest 3の初期設定にも使う[Meta Horizonモバイルアプリアプリ](https://www.meta.com/ja-jp/help/quest/articles/getting-started/getting-started-with-quest-3/set-up-quest-3/)。

あと。
ARデバイスを使って屋外で仮想でドローンを飛ばせたら、おもしろいことができるのかなとも思ってます。
Quest 3のセットアップのページにもありましたが直射日光を避けるようにとありますので、日陰や北側窓内側から操作するとか、なんらかの対策いけそうだとか。

**Reply by tmori** [reply]
> はい。この手のものをせっかく入手するなら別機種にして動作検証して対応機種のバリエーションを増やせたらいいかなと思ったりも。

いいですね。現時点で、Quest3Sは、すでにお試しいただく方が現れていますね。
Pico4 ultraは初めて知りました。Unityで開発できるみたいなので、箱庭のりそうです。

ただ、現状の箱庭のARコードは Quest3のライブラリ(Oculus Integration)を使っていますので、別機種ですと、それと同等の機能での対応になるかなとは思います。（多分、同じような機能はあるとは思っております）

> デモ映像を撮られたときは、wifiでしょうか？

これは、どちらでもいけます。
Wifiでつまづいたのは、電子レンジがある部屋でやった時ですね。なので、イベントとかだと怖いのでLANにすることにしました。

電波干渉がない環境ならWifiでいけると思います。
（UDP通信で、1kb未満のデータを60msec単位で互いに送信しているだけです。）

> スマホも必要ですね。

ですね。Quest3の場合は、初期設定で使いますね。

> ARデバイスを使って屋外で仮想でドローンを飛ばせたら、おもしろい

これはそう。その通り！

**Reply by yamati-kz** [reply]
> ただ、現状の箱庭のARコードは Quest3のライブラリ(Oculus Integration)を使っていますので、別機種ですと、それと同等の機能での対応になるかなとは思います。（多分、同じような機能はあるとは思っております

今後は[Oculus Integration (Deprecated)](https://assetstore.unity.com/packages/tools/integration/oculus-integration-deprecated-82022?locale=ja-JP&srsltid=AfmBOorFJMXeOOtlMmxA_E39beLt9vh_SSMRKvQeQdnSLB1Y0-XFgc-4)に代って、[Meta XR All-in-One SDK](https://assetstore.unity.com/packages/tools/integration/meta-xr-all-in-one-sdk-269657?locale=ja-JP&srsltid=AfmBOor5iyy79cF3pQ5mrXGf2muNPVZsTdGF5g8NBU5EndA3CT2nihAS)に移られていくのかなと想像しております。
[AssetStore](https://assetstore.unity.com/)でPico4で検索してみましたがQuest3のようなライブラリが見つかりませんでした。
ですのでの購入候補から一旦除外ですね。

**Reply by yamati-kz** [reply]
失礼しました。[PICO-Unity-Integration-SDK](https://github.com/Pico-Developer/PICO-Unity-Integration-SDK)を見つけました。

**Comment by tmori** [comment]
> 2\. ルータ性能の影響

多分、最近のルーターならなんでもいけると思います。自分は、Wi-Fi6対応のものを使っております。

いちよ型番も書いておきます。

ELECOM WRC-X1500GS-B

**Reply by yamati-kz** [reply]
ルータ情報、ありがとうございます。
私の手持ちの[WR8700N](https://www.aterm.jp/product/atermstation/product/warpstar/wr8700n-hp/spec.html)が、かなり古いので、どうしようかなと。
有線は1000BASE-Tなのでwifiの遅さが耐えられなかったら有線LANのお試しするのも、それなりにいいかなとも思ったりしてます。

**Reply by tmori** [reply]
1000BASE-Tなら、多分、大丈夫ではないかなーっと思います。

**Reply by yamati-kz** [reply]
[WRH-583YL2-S](https://www.elecom.co.jp/products/WRH-583YL2-S.html)でも古いですが試してみます。

**Comment by yamati-kz** [comment]
@tmori さん いろいろと教えていただき、ありがとうございました。

私の場合、ARデバイスの用途を箱庭ドローンを中心に考えるとQuest 3Sで十分ではと思うのですが、[フレネルレンズ](https://lookaside.fbsbx.com/elementpath/media/?media_id=1633019700763467&version=1723761780&transcode_extension=webp)が気に入らないといいますか... 廉価版なのてしかたないですが。
見やすさを気にするところがあるので、Quest 3かなとも。128G版の販売がなくなったのが痛いですね。512G版が値下げされてますが。

ARデバイスは試すつもりですので、別途レポートさせていただければと思います。
